  ---
title: "nucleotide_diversity_HW"
author: "Max McClure"
date: "2025-09-23"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE) 
knitr::opts_knit$set(root.dir = "~/projects/eco_genomics_2025/population_genomics/myresults/homework1results")
```

Import our libraries for plotting

```{r}
library(ggplot2)
library(tidyverse)
library(ggpubr)
```

read in theta outputs

```{r}
theta_ST0 <- read.table("2032_ALL_ST0_win50000_step50000.thetas", header = TRUE, sep = "\t")

theta_ST1 <- read.table("2032_ALL_ST1_win50000_step50000.thetas", header = TRUE, sep = "\t")

theta_SMDI1 <- read.table("2032_ALL_SMDI1_win50000_step50000.thetas", header = TRUE, sep = "\t")

theta_SMDI2 <- read.table("2032_ALL_SMDI2_win50000_step50000.thetas", header = TRUE, sep = "\t")

theta_ST0_SMDI2 <- read.table("2032_ALL_ST0_SMDI2_win50000_step50000.thetas", header = TRUE, sep = "\t")
```

```{r}
theta_ST0$tWsite = theta_ST0$tW/theta_ST0$nSites
theta_ST0$tPsite = theta_ST0$tP/theta_ST0$nSites

theta_ST1$tWsite = theta_ST1$tW/theta_ST1$nSites
theta_ST1$tPsite = theta_ST1$tP/theta_ST1$nSites

theta_SMDI1$tWsite = theta_SMDI1$tW/theta_SMDI1$nSites
theta_SMDI1$tPsite = theta_SMDI1$tP/theta_SMDI1$nSites

theta_SMDI2$tWsite = theta_SMDI2$tW/theta_SMDI2$nSites
theta_SMDI2$tPsite = theta_SMDI2$tP/theta_SMDI2$nSites

theta_ST0_SMDI2$tWsite = theta_ST0_SMDI2$tW/theta_ST0_SMDI2$nSites
theta_ST0_SMDI2$tPsite = theta_ST0_SMDI2$tP/theta_ST0_SMDI2$nSites
```

```{r}
theta2_ST0 <- theta_ST0 %>%
  arrange(Chr, WinCenter) %>%
  filter(nSites > 100)

theta2_ST1 <- theta_ST1 %>%
  arrange(Chr, WinCenter) %>%
  filter(nSites > 100)

theta2_SMDI1 <- theta_SMDI1 %>%
  arrange(Chr, WinCenter) %>%
  filter(nSites > 100)

theta2_SMDI2 <- theta_SMDI2 %>%
  arrange(Chr, WinCenter) %>%
  filter(nSites > 100)

theta2_ST0_SMDI2 <- theta_ST0_SMDI2 %>%
  arrange(Chr, WinCenter) %>%
  filter(nSites > 100)
```

plotting histogram of total number of sites are in each seq window
```{r}
SitesPerWindowST0 <- ggplot(data = theta2_ST0, aes(x = nSites)) +
  geom_histogram(bins = 50, fill = "lightblue", color = "black") +
  geom_vline(xintercept = median(theta2_ST0$nSites), color = "red", linetype = "dashed") #+
   #labs(title = "Distribution of windows with data", x = "Number of sites in 50kb window", y = "Frequency")


SitesPerWindowST1 <- ggplot(data = theta2_ST1, aes(x = nSites)) +
  geom_histogram(bins = 50, fill = "lightblue", color = "black") +
  geom_vline(xintercept = median(theta2_ST1$nSites), color = "red", linetype = "dashed") 

SitesPerWindowSMDI1 <- ggplot(data = theta2_SMDI1, aes(x = nSites)) +
  geom_histogram(bins = 50, fill = "lightblue", color = "black") +
  geom_vline(xintercept = median(theta2_SMDI1$nSites), color = "red", linetype = "dashed") 

SitesPerWindowSMDI2 <- ggplot(data = theta2_SMDI2, aes(x = nSites)) +
  geom_histogram(bins = 50, fill = "lightblue", color = "black") +
  geom_vline(xintercept = median(theta2_SMDI2$nSites), color = "red", linetype = "dashed") 

SitesPerWindowST0_SMDI2 <- ggplot(data = theta2_ST0_SMDI2, aes(x = nSites)) +
  geom_histogram(bins = 50, fill = "lightblue", color = "black") +
  geom_vline(xintercept = median(theta2_ST0_SMDI2$nSites), color = "red", linetype = "dashed") 

ggarrange(SitesPerWindowST0, SitesPerWindowST1, 
          SitesPerWindowSMDI1, SitesPerWindowSMDI2,
          SitesPerWindowST0_SMDI2,
          labels = "AUTO"
          )
?ggarrange
```

plot pi

```{r}

piST0 <- ggplot(theta2_ST0, aes(x=Chr, y = tPsite, color = Chr)) +
  geom_point(size = 0.75, shape = 8, show.legend = FALSE) +
  theme_classic() +
  labs(x = "Contig", y = "Theta Pi") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) +
  ylim(0,0.055) #+
  #labs(title = "Nucleotide diversity in 50kb windows for population 2032", x = "Scaffold", y = "Theta-pi")

piST1 <- ggplot(theta2_ST1, aes(x=Chr, y = tPsite, color = Chr)) +
  geom_point(size = 0.75, shape = 8, show.legend = FALSE) + 
  theme_classic() +
  labs(x = "Contig", y = "Theta Pi") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) +
  ylim(0,0.055) 

piSMDI1 <- ggplot(theta2_SMDI1, aes(x=Chr, y = tPsite, color = Chr)) +
  geom_point(size = 0.75, shape = 8, show.legend = FALSE) +
  theme_classic() +
  labs(x = "Contig", y = "Theta Pi") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) +
  
  ylim(0,0.055)


piSMDI2 <- ggplot(theta2_SMDI2, aes(x=Chr, y = tPsite, color = Chr)) +
  geom_point(size = 0.75, shape = 8, show.legend = FALSE) +
  theme_classic() +
  labs(x = "Contig", y = "Theta Pi") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) +
  ylim(0,0.055) 

piST0_SMDI2 <- ggplot(theta2_ST0_SMDI2, aes(x=Chr, y = tPsite, color = Chr)) +
  geom_point(size = 0.75, shape = 8, show.legend = FALSE) +
  theme_classic() +
  labs(x = "Contig", y = "Theta Pi") +
  theme(axis.text.x = element_blank(),
        axis.ticks.x = element_blank()) +
  ylim(0,0.055) 
piST0_SMDI2
ggarrange(piST0, piST1, piSMDI1, piSMDI2, piST0_SMDI2,
          labels = "AUTO")
```


```{r}
lmST0 <- lm(tPsite ~ tWsite, data = theta2_ST0)
lmST1 <- lm(tPsite ~ tWsite, data = theta2_ST1)
lmSMDI1 <- lm(tPsite ~ tWsite, data = theta2_SMDI1)
lmSMDI2 <- lm(tPsite ~ tWsite, data = theta2_SMDI2)
lmST0_SMDI2 <- lm(tPsite ~ tWsite, data = theta2_ST0_SMDI2)

lms <- data.frame(setting = c("ST0","ST1", "SMDI1","SMDI2","ST0_SMDI2"),
                   slope = c(lmST0 = lmST0$coefficients[2], 
                             lmST1 = lmST1$coefficients[2],
                             lmSMDI1 = lmSMDI1$coefficients[2], 
                             lmSMDI2 = lmSMDI2$coefficients[2],
                             lmST0_SMDI2 = lmST0_SMDI2$coefficients[2]))


WvPiST0 <- ggplot(theta2_ST0, aes(x = tWsite, y = tPsite, color = Chr)) +
  geom_point(size = 0.75, shape = 8, show.legend = FALSE) +
  labs(x = "theta-W", y = "theta-pi") + 
  geom_abline(slope = 1, intercept = 0, color = "black") +
  geom_abline(slope = lms$slope[1], intercept = 0, 
              linetype = "dashed", color = "red") +
  theme_classic()
WvPiST0
?geom_abline
WvPiST1 <- ggplot(theta2_ST1, aes(x = tWsite, y = tPsite, color = Chr)) +
  geom_point(size = 0.75, shape = 8, show.legend = FALSE) +
  labs(x = "theta-W", y = "theta-pi") + 
  geom_abline(slope = 1, intercept = 0, color = "black") +
  geom_abline(slope = lms$slope[2], intercept = 0, 
              linetype = "dashed", color = "red") +
  theme_classic()

WvPiSMDI1 <- ggplot(theta2_SMDI1, aes(x = tWsite, y = tPsite, color = Chr)) +
  geom_point(size = 0.75, shape = 8, show.legend = FALSE) +
  labs(x = "theta-W", y = "theta-pi") + 
  geom_abline(slope = 1, intercept = 0, color = "black") +
  geom_abline(slope = lms$slope[3], intercept = 0, 
              linetype = "dashed", color = "red") +
  theme_classic()

WvPiSMDI2 <- ggplot(theta2_SMDI2, aes(x = tWsite, y = tPsite, color = Chr)) +
  geom_point(size = 0.75, shape = 8, show.legend = FALSE) +
  labs(x = "theta-W", y = "theta-pi") + 
  geom_abline(slope = 1, intercept = 0, color = "black") +
  geom_abline(slope = lms$slope[4], intercept = 0, 
              linetype = "dashed", color = "red") +
  theme_classic()

WvPiST0_SMDI2 <- ggplot(theta2_ST0_SMDI2, aes(x = tWsite, y = tPsite, color = Chr)) +
  geom_point(size = 0.75, shape = 8, show.legend = FALSE) +
  labs(x = "theta-W", y = "theta-pi") + 
  geom_abline(slope = 1, intercept = 0, color = "black") +
  geom_abline(slope = lms$slope[5], intercept = 0, 
              linetype = "dashed", color = "red") +
  theme_classic()

ggarrange(WvPiST0, WvPiST1, WvPiSMDI1, WvPiSMDI2, WvPiST0_SMDI2,
          labels = "AUTO")


```


which windows have evidence of negative D and low nucleotide diversity ?
 - looking for selective sweep (demographic change), entire genome alteration
```{r}
theta2_ST0[which(theta2_ST0$Tajima<-1.5 & theta2_ST0$tPsite < 0.001),]
theta2_ST1[which(theta2_ST1$Tajima<-1.5 & theta2_ST1$tPsite < 0.001),]
theta2_SMDI1[which(theta2_SMDI1$Tajima<-1.5 & theta2_SMDI1$tPsite < 0.001),]
theta2_SMDI2[which(theta2_SMDI2$Tajima<-1.5 & theta2_SMDI2$tPsite < 0.001),]
theta2_ST0_SMDI2[which(theta2_ST0_SMDI2$Tajima<-1.5 & theta2_ST0_SMDI2$tPsite < 0.001),]


summary(theta2_ST0)
```





```{r}

library(gt)

stats <- function(theta2){
  mu = 2.5*10^-9
  gen.time = 30
  C = 4
  mean_tW <- mean(theta2$tWsite)
  mean_tP <- mean(theta2$tPsite)
  Ne_theta <- mean_tW / (mu*gen.time*C)
  Ne_pi <- mean_tP / (mu*gen.time*C)
  maxD = max(theta2$Tajima)
  minD = min(theta2$Tajima)
  BigD = theta2[which(theta2$Tajima == maxD),]
  SmallD =  theta2[which(theta2$Tajima == minD),]
  mean_Tajima <- mean(theta2$Tajima)
  return(list(mean_tW = mean_tW, mean_tP = mean_tP, 
              Ne_theta = Ne_theta, Ne_pi = Ne_pi,
              maxD = maxD, minD = minD,
              meanTajima = mean_Tajima,
              BigD = BigD$Chr, SmallD = SmallD$Chr))
}

ST0stats <- stats(theta2_ST0)
ST1stats <- stats(theta2_ST1)
SMDI1stats <- stats(theta2_SMDI1)
SMDI2stats <- stats(theta2_SMDI2)
ST0_SMDI2stats <- stats(theta2_ST0_SMDI2)


table_data <- tibble(
  a = c("skipTriallelic 0", "skipTriallelic 1", 
            "setMinDepthInd 1", "setMinDepthInd 2", 
            "skipTriallelic 1 & setMinDepthInd 2"),
  Average_tW = c(ST0stats$mean_tW, ST1stats$mean_tW,
                SMDI1stats$mean_tW, SMDI2stats$mean_tW, 
                ST0_SMDI2stats$mean_tW),
  Average_tP = c(ST0stats$mean_tP, ST1stats$mean_tP,
                SMDI1stats$mean_tP, SMDI2stats$mean_tP, 
                ST0_SMDI2stats$mean_tP),
  Ne_theta = c(ST0stats$Ne_theta, ST1stats$Ne_theta,
                SMDI1stats$Ne_theta, SMDI2stats$Ne_theta, 
                ST0_SMDI2stats$Ne_theta),
  Ne_pi = c(ST0stats$Ne_pi, ST1stats$Ne_pi,
                SMDI1stats$Ne_pi, SMDI2stats$Ne_pi, 
                ST0_SMDI2stats$Ne_pi),
  maxD = c(ST0stats$maxD, ST1stats$maxD,
                SMDI1stats$maxD, SMDI2stats$maxD, 
                ST0_SMDI2stats$maxD),
  minD = c(ST0stats$minD, ST1stats$minD,
                SMDI1stats$minD, SMDI2stats$minD, 
                ST0_SMDI2stats$minD),
  meanTajima = c(ST0stats$meanTajima, ST1stats$meanTajima,
                SMDI1stats$meanTajima, SMDI2stats$meanTajima, 
                ST0_SMDI2stats$meanTajima),
  slope = c(lms$slope[1], lms$slope[2], lms$slope[3], lms$slope[4], lms$slope[5]))

gtable <- table_data %>%
  gt() %>%
  tab_header(
    title = "Diversity Statistics between ANGSD settings") %>%
  cols_label( 
    a = "Altered Setting",
    Average_tW = "Mean theta W",
    Average_tP = "Mean theta Pi",
    Ne_theta = "Theta W Effective population size",
    Ne_pi = "Theta Pi Effective population size",
    maxD = "Maximum Tajima's D",
    minD = "Minimum Tajima's D",
    meanTajima = "Mean Tajima's D", 
    slope = "Slope of Theta W by Theta Pi"
    ) %>% 
  fmt_auto(table_data, 
           columns = everything(), rows = everything(),
           lg_num_pref = c("sci")) %>%
  tab_options(
    heading.align = "left"
  )
?gt
gtable


```












